cssom-view-1/Overview.bs

```diff
@@ -1070,21 +1070,16 @@ result of running these steps:
         <dt><a>caret range</a>
         <dd>null
     </dl>
-1. Otherwise, return a <a>caret position</a> where the
-    <a>caret range</a> is a collapsed
-    {{Range}} object for the position
-    where the text insertion point indicator would have been inserted when applying the <a>transforms</a> that apply to the descendants of the
-    <a>viewport</a>, and the other properties are set as follows:
-
-    <dl>
-        <dt><a>caret node</a>
-        <dd>The [=range/start node=]
-            of the <a>caret range</a>.
-
-        <dt><a>caret offset</a>
-        <dd>The [=range/start offset=] of
-            the <a>caret range</a>.
-    </dl>
+1. Otherwise, return a <a>caret position</a> where its properties are set after running these steps:
+
+    1. Let <a>caret range</a> to be a collapsed {{Range}} object for the position where the text insertion point indicator whould have been inserted when applying
+        the <a>transforms</a> that apply to the descendants of the <a>viewport</a>.
+    1. Let <var>startNode</var> be the [=range/start node=] of the <a>caret range</a>, and let <var>startOffset</var> be the [=range/start offset=] of the <a>caret range</a>.
+    1. While <var>startNode</var> is a [=node=], <var>startNode</var>'s [=tree/root=] is a [=shadow root=], and <var>startNode</var>'s [=tree/root=] is not a [=shadow-including inclusive ancestor=] of any of <var>shadowRoots</var>, repeat these steps:
+        1. Set <var>startOffset</var> to [=tree/index=] of <var>startNode</var>'s [=tree/root=]'s [=host=].
+        1. Set <var>startNode</var> to <var>startNode</var>'s [=tree/root=]'s [=host=]'s [=tree/parent=].</li>
+    1. Set <a>caret node</a> in <a>caret position</a> to <var>startNode</var>, and set <a>caret offset</a> in <a>caret position</a> to <var>startOffset</var>.
+    1. Set <a>caret range</a>'s [=range/start node=] and [=range/end node=] to <var>startNode</var>, and set <a>caret range</a>'s [=range/start offset=] and [=range/end offset=] to <var>startOffset</var>.
```

Comment:
nit: it might be more straightforward to have a `<var>caretRange</var>` that's declared similarly to `<var>startNode</var>` and `<var>startOffset</var>`, and have that be the thing that is manipulated in these steps. Then you could have a single last step that would set all 3 things on the `CaretPosition`:
`1. Set the <a>caret position</a>'s <a>caret node</a> to <var>startNode</var>, <a>caret offset</a> to <var>startOffset</var>, and <a>caret range</a> to <var>caretRange</var>.`

Reply:
Going further with that approach, you could eliminate the earlier bit that says "...return a <a>caret position</a> where its properties are set after running these steps:", and instead have the last step be:

`1. Return a <a>caret position</a> whose <a>caret node</a> is set to <var>startNode</var>, <a>caret offset</a> is set to <var>startOffset</var>, and <a>caret range</a> is set to <var>caretRange</var>.`

Reply:
Updated in next iteration. Now the single last step sets all three things on `CaretPosition`.